src/main/java/apoc/schema/Schemas.java

```diff
@@ -117,6 +135,101 @@ private AssertSchemaResult createUniqueConstraint(Schema schema, String lbl, Str
+                IndexDefinition index = it.next();
+                if (!index.isConstraintIndex()) {
+                    idx.add(index);
+                }
+            }
+            return idx;
+        }).get(allIndexesTimeout, TimeUnit.MILLISECONDS);
+
+        Util.inTxFuture(executorService, db, () -> {
+            for (IndexDefinition definition : indexesToRecreate) {
+                String label = definition.getLabel().name();
+                List<String> keys = new ArrayList<>();
+                definition.getPropertyKeys().forEach(keys::add);
+
+                AssertSchemaResult info = new AssertSchemaResult(label, keys);
+
+                definition.drop();
+
+                info.dropped();
+                result.add(info);
+            }
+            return null;
+        }).get(dropAllIndexesTimeout,TimeUnit.MILLISECONDS);
+
+        Util.inTxFuture(executorService, db, () -> {
+            for (IndexDefinition definition : indexesToRecreate) {
+                List<String> propertyKeys = new ArrayList<>();
+                for (String propertyKey : definition.getPropertyKeys()) {
+                    propertyKeys.add(propertyKey);
+                }
+
+                if (propertyKeys.size() == 1) {
+                    result.add(createSinglePropertyIndex(schema, definition.getLabel().name(), propertyKeys.get(0)));
+
+                } else {
+                    result.add(createCompoundIndex(definition.getLabel().name(), propertyKeys));
+                }
+            }
+            return null;
+        }).get(recreateAllIndexesTimeout, TimeUnit.MILLISECONDS);
+
+        return result;
+    }
+
+    public List<AssertSchemaResult> recreateConstraints(long allConstraintsTimeout, long dropAllConstraintsTimeout, long recreateAllConstraintsTimeout) throws ExecutionException, InterruptedException, TimeoutException {
+        Schema schema = db.schema();
+        List<AssertSchemaResult> result = Collections.synchronizedList(new ArrayList<>());
+        ExecutorService executorService = Pools.SINGLE;
+        List<ConstraintDefinition> constraintsToRecreate = Util.inTxFuture(executorService, db, () -> {
+            List<ConstraintDefinition> idx = new ArrayList<>();
+            Iterable<ConstraintDefinition> outerIndexes = schema.getConstraints();
+            for (ConstraintDefinition index : outerIndexes) {
+                idx.add(index);
+            }
+            return idx;
+        }).get(allConstraintsTimeout, TimeUnit.MILLISECONDS);
+
+        Util.inTxFuture(executorService, db, () -> {
+            for (ConstraintDefinition constraint : constraintsToRecreate) {
+                String label = constraint.isConstraintType(ConstraintType.RELATIONSHIP_PROPERTY_EXISTENCE) ? constraint.getRelationshipType().name() : constraint.getLabel().name();
+                AssertSchemaResult info = new AssertSchemaResult(label, Iterables.asList(constraint.getPropertyKeys())).unique();
```

Comment:
third variant of transforming `Iterables.asList(constraint.getPropertyKeys())` into a List :) we should pick one and stay consistent.